// ubl-builder/lib/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UdtTypes = exports.Address = exports.AddressLine = exports.CctBinaryObjectType = exports.CctAmountType = exports.Invoice = void 0;
var cct_1 = require("./ubl21/types/UnqualifiedDataTypes/essentials/cct");
Object.defineProperty(exports, "CctAmountType", { enumerable: true, get: function () { return cct_1.CctAmountType; } });
Object.defineProperty(exports, "CctBinaryObjectType", { enumerable: true, get: function () { return cct_1.CctBinaryObjectType; } });
var AddressLine_1 = require("./ubl21/CommonAggregateComponents/AddressLine");
Object.defineProperty(exports, "AddressLine", { enumerable: true, get: function () { return AddressLine_1.AddressLine; } });
var AddressTypeGroup_1 = require("./ubl21/CommonAggregateComponents/AddressTypeGroup");
Object.defineProperty(exports, "Address", { enumerable: true, get: function () { return AddressTypeGroup_1.Address; } });
var UdtTypes = require("./ubl21/types/UnqualifiedDataTypes");
exports.UdtTypes = UdtTypes;
var schemaDocuments_1 = require("./ubl21/schemaDocuments");
Object.defineProperty(exports, "Invoice", { enumerable: true, get: function () { return schemaDocuments_1.Invoice; } });
