// ubl-builder/lib/ubl21/extensionComponents/UBLExtension.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.UBLExtensionType = void 0;
var GenericAggregateComponent_1 = require("../CommonAggregateComponents/GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var AnyExtensionContent_1 = require("./AnyExtensionContent");
/*
    1    cbc:ID [0..1]     An identifier for the Extension assigned by the creator of the extension.
    2    cbc:Name [0..1]     A name for the Extension assigned by the creator of the extension.
    3    ext:ExtensionAgencyID [0..1]     An agency that maintains one or more Extensions.
    4    ext:ExtensionAgencyName [0..1]     The name of the agency that maintains the Extension.
    5    ext:ExtensionVersionID [0..1]     The version of the Extension.
    6    ext:ExtensionAgencyURI [0..1]     A URI for the Agency that maintains the Extension.
    7    ext:ExtensionURI [0..1]     A URI for the Extension.
    8    ext:ExtensionReasonCode [0..1]     A code for reason the Extension is being included.
    9    ext:ExtensionReason [0..1]     A description of the reason for the Extension.
    10    ext:ExtensionContent [1..1]     The definition of the extension content.
*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    name: { order: 2, attributeName: 'cbc:Name', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtName },
    extensionAgencyID: { order: 3, attributeName: 'ext:ExtensionAgencyID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    extensionAgencyName: { order: 4, attributeName: 'ext:ExtensionAgencyName', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    extensionVersionID: { order: 5, attributeName: 'ext:ExtensionVersionID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    extensionAgencyURI: { order: 6, attributeName: 'ext:ExtensionAgencyURI', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    extensionURI: { order: 7, attributeName: 'ext:ExtensionURI', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    extensionReasonCode: { order: 8, attributeName: 'ext:ExtensionReasonCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    extensionReason: { order: 9, attributeName: 'ext:ExtensionReason', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    extensionContent: { order: 10, attributeName: 'ext:ExtensionContent', min: 1, max: 1, classRef: AnyExtensionContent_1.default },
};
/**
 *
 */
var UBLExtension = /** @class */ (function (_super) {
    __extends(UBLExtension, _super);
    /**     *
     * @param {AllowedParams} content
     * @param {string} name
     */
    function UBLExtension(content) {
        return _super.call(this, content, ParamsMap, 'cac:UBLExtension') || this;
    }
    UBLExtension.prototype.setExtensionContent = function (value) {
        if (!(value instanceof AnyExtensionContent_1.default)) {
            throw new Error('value must be an AnyExtensionContent instance');
        }
        this.attributes.extensionContent = value;
    };
    /**
     * @returns {AnyExtensionContent}
     */
    UBLExtension.prototype.getExtensionContent = function () {
        return this.attributes.extensionContent;
    };
    return UBLExtension;
}(GenericAggregateComponent_1.default));
exports.UBLExtensionType = UBLExtension;
