// ubl-builder/lib/ubl21/types/UnqualifiedDataTypes/essentials/cct/CctBinaryObject.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CctBinaryObjectType = void 0;
var xsd_1 = require("../xsd");
/**
 * cct:BinaryObjectType
 *  A set of finite-length sequences of binary octets.
 *
 * Namespace: urn:un:unece:uncefact:data:specification:CoreComponentTypeSchemaModule:2
 *
 * CCTS properties: *
 *   Unique ID: UNDT000002
 *   Category Code: CCT
 *   Dictionary Entry Name: Binary Object. Type
 *   Version I D: 1.0
 *   Definition: A set of finite-length sequences of binary octets.
 *   Representation Term Name: Binary Object
 *   Primitive Type: binary
 *
 * MORE INFO: http://www.datypic.com/sc/ubl21/t-cct_BinaryObjectType.html
 */
var CctBinaryObjectType = /** @class */ (function (_super) {
    __extends(CctBinaryObjectType, _super);
    /**
     *
     * @param content
     * @param attributes
     */
    function CctBinaryObjectType(content, attributes) {
        return _super.call(this, content, attributes) || this;
    }
    CctBinaryObjectType.prototype.parseToJson = function () {
        var _this = this;
        var jsonResult = { '#text': this.content };
        Object.keys(this.attributes)
            .filter(function (att) { return _this.attributes[att]; })
            .forEach(function (attribute) {
            jsonResult["@" + attribute] = _this.attributes[attribute];
        });
        return jsonResult;
    };
    /**
     * @param value
     */
    CctBinaryObjectType.prototype.setFormat = function (value) {
        this.attributes.format = value;
    };
    /**
     *
     * @param {string} value
     */
    CctBinaryObjectType.prototype.setMimecode = function (value) {
        this.attributes.mimeCode = value;
    };
    /**
     *
     * @param {string} value
     */
    CctBinaryObjectType.prototype.setEncodingCode = function (value) {
        this.attributes.encodingCode = value;
    };
    CctBinaryObjectType.prototype.setCharacterSetCode = function (value) {
        this.attributes.characterSetCode = value;
    };
    /**
     *
     * @param {string} value
     */
    CctBinaryObjectType.prototype.setUri = function (value) {
        this.attributes.uri = value;
    };
    CctBinaryObjectType.prototype.setFileName = function (value) {
        this.attributes.filename = value;
    };
    return CctBinaryObjectType;
}(xsd_1.XsdNormalizedString));
exports.CctBinaryObjectType = CctBinaryObjectType;
