// ubl-builder/lib/ubl21/types/UnqualifiedDataTypes/essentials/cct/CctCode.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CctCodeType = void 0;
var xsd_1 = require("../xsd");
/**
 * A character string to identify and distinguish uniquely,
 * one instance of an object in an identification scheme from all other objects in
 * the same scheme together with relevant supplementary information.
 */
var CctCodeType = /** @class */ (function (_super) {
    __extends(CctCodeType, _super);
    /**
     *
     * @param content
     * @param attributes
     */
    function CctCodeType(content, attributes) {
        return _super.call(this, content, attributes) || this;
    }
    CctCodeType.prototype.parseToJson = function () {
        var _this = this;
        var jsonResult = { '#text': this.content };
        Object.keys(this.attributes)
            .filter(function (att) { return _this.attributes[att]; })
            .forEach(function (attribute) {
            jsonResult["@" + attribute] = _this.attributes[attribute];
        });
        return jsonResult;
    };
    CctCodeType.prototype.setListID = function (value) {
        this.attributes.listID = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setListAgencyID = function (value) {
        this.attributes.listAgencyID = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setListAgencyName = function (value) {
        this.attributes.listAgencyName = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setListName = function (value) {
        this.attributes.listName = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setlistVersionID = function (value) {
        this.attributes.listVersionID = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setName = function (value) {
        this.attributes.name = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setLanguageID = function (value) {
        this.attributes.languageID = value;
    };
    CctCodeType.prototype.setListUri = function (value) {
        this.attributes.listURI = value;
    };
    /** @param {string} value */
    CctCodeType.prototype.setListSchemeURI = function (value) {
        this.attributes.listSchemeURI = value;
    };
    return CctCodeType;
}(xsd_1.XsdNormalizedString));
exports.CctCodeType = CctCodeType;
