// ubl-builder/lib/ubl21/types/UnqualifiedDataTypes/essentials/cct/CctQuantity.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CctQuantityType = void 0;
var xsd_1 = require("../xsd");
/**
 *
 * Unique I D: UNDT000018
 * Category Code: CCT
 * Dictionary Entry Name: Quantity. Type
 * Version I D: 1.0
 * Definition: A counted number of non-monetary units possibly including fractions.
 * Representation Term Name: Quantity
 * Primitive Type: decimal
 *
 * A counted number of non-monetary units possibly including fractions.
 * More info: http://www.datypic.com/sc/ubl21/t-cct_QuantityType.html
 */
var CctQuantityType = /** @class */ (function (_super) {
    __extends(CctQuantityType, _super);
    function CctQuantityType(content, attributes) {
        return _super.call(this, content, attributes) || this;
    }
    CctQuantityType.prototype.validateContent = function () {
        _super.prototype.validateContent.call(this);
    };
    CctQuantityType.prototype.parseToJson = function () {
        var _this = this;
        var jsonResult = { '#text': this.content };
        Object.keys(this.attributes)
            .filter(function (att) { return _this.attributes[att]; })
            .forEach(function (att) {
            jsonResult["@" + att] = _this.attributes[att];
        });
        return jsonResult;
    };
    CctQuantityType.prototype.setUnitCode = function (value) {
        this.attributes.unitCode = value;
    };
    CctQuantityType.prototype.setUnitCodeListID = function (value) {
        this.attributes.unitCodeListID = value;
    };
    CctQuantityType.prototype.setUnitCodeListAgencyID = function (value) {
        this.attributes.unitCodeListAgencyID = value;
    };
    CctQuantityType.prototype.setUnitCodeListAgencyName = function (value) {
        this.attributes.unitCodeListAgencyName = value;
    };
    return CctQuantityType;
}(xsd_1.XsdDecimal));
exports.CctQuantityType = CctQuantityType;
