// ubl-builder/lib/ubl21/types/UnqualifiedDataTypes/essentials/cct/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CctMeasureType = exports.CctQuantityType = exports.CctNumericType = exports.CctIdentifierType = exports.CctCodeType = exports.CctBinaryObjectType = exports.CctAmountType = void 0;
/**
 * more info http://www.datypic.com/sc/ubl21/s-CCTS_CCT_SchemaModule-2.1.xsd.html
 */
var CctAmount_1 = require("./CctAmount");
Object.defineProperty(exports, "CctAmountType", { enumerable: true, get: function () { return CctAmount_1.CctAmountType; } });
var CctBinaryObject_1 = require("./CctBinaryObject");
Object.defineProperty(exports, "CctBinaryObjectType", { enumerable: true, get: function () { return CctBinaryObject_1.CctBinaryObjectType; } });
var CctCode_1 = require("./CctCode");
Object.defineProperty(exports, "CctCodeType", { enumerable: true, get: function () { return CctCode_1.CctCodeType; } });
var CctIdentifier_1 = require("./CctIdentifier");
Object.defineProperty(exports, "CctIdentifierType", { enumerable: true, get: function () { return CctIdentifier_1.CctIdentifierType; } });
var CctNumeric_1 = require("./CctNumeric");
Object.defineProperty(exports, "CctNumericType", { enumerable: true, get: function () { return CctNumeric_1.CctNumericType; } });
var CctQuantity_1 = require("./CctQuantity");
Object.defineProperty(exports, "CctQuantityType", { enumerable: true, get: function () { return CctQuantity_1.CctQuantityType; } });
var CctMeasure_1 = require("./CctMeasure");
Object.defineProperty(exports, "CctMeasureType", { enumerable: true, get: function () { return CctMeasure_1.CctMeasureType; } });
