// ubl-builder/lib/ubl21/extensionComponents/customExtensionContents/AuthorizationProvider.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizationProvider = void 0;
var GenericAggregateComponent_1 = require("../../CommonAggregateComponents/GenericAggregateComponent");
var UdtIdentifier_1 = require("../../types/UnqualifiedDataTypes/UdtIdentifier");
var ParamsMap = {
    authorizationProviderID: {
        order: 1,
        attributeName: 'sts:AuthorizationProviderID',
        min: 0,
        max: 1,
        classRef: UdtIdentifier_1.UdtIdentifier,
    },
};
/**
 * Body of Dian extension content
 */
var AuthorizationProvider = /** @class */ (function (_super) {
    __extends(AuthorizationProvider, _super);
    /**     *
     * @param {AllowedParams} content
     * @param {string} name
     */
    function AuthorizationProvider(content) {
        return _super.call(this, content, ParamsMap, 'sts:AuthorizationProvider') || this;
    }
    return AuthorizationProvider;
}(GenericAggregateComponent_1.default));
exports.AuthorizationProvider = AuthorizationProvider;
