// ubl-builder/lib/ubl21/extensionComponents/customExtensionContents/DianExtensionContent.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DianExtensionsContent = void 0;
var GenericAggregateComponent_1 = require("../../CommonAggregateComponents/GenericAggregateComponent");
var InvoiceControl_1 = require("./InvoiceControl");
var UnqualifiedDataTypes_1 = require("../../types/UnqualifiedDataTypes");
var InvoiceSource_1 = require("./InvoiceSource");
var SoftwareProvider_1 = require("./SoftwareProvider");
var AuthorizationProvider_1 = require("./AuthorizationProvider");
var ParamsMap = {
    // ##################################  TODO CAC MISSING ################################################
    invoiceControl: { order: 1, attributeName: 'sts:InvoiceControl', min: 0, max: 1, classRef: InvoiceControl_1.InvoiceControl },
    invoiceSource: { order: 2, attributeName: 'sts:InvoiceSource', min: 0, max: 1, classRef: InvoiceSource_1.InvoiceSource },
    softwareProvider: { order: 3, attributeName: 'sts:SoftwareProvider', min: 0, max: 1, classRef: SoftwareProvider_1.SoftwareProvider },
    softwareSecurityCode: {
        order: 4,
        attributeName: 'sts:SoftwareSecurityCode',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    authorizationProvider: {
        order: 5,
        attributeName: 'sts:AuthorizationProvider',
        min: 0,
        max: 1,
        classRef: AuthorizationProvider_1.AuthorizationProvider,
    },
    QRCode: { order: 6, attributeName: 'sts:QRCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
};
/**
 * Contenido de las definiciónes de extensions de la DIAN.
 */
var DianExtensionsContent = /** @class */ (function (_super) {
    __extends(DianExtensionsContent, _super);
    /**     *
     * @param {AllowedParams} content
     * @param {string} name
     */
    function DianExtensionsContent(content) {
        return _super.call(this, content, ParamsMap, 'cac:DianExtensions') || this;
    }
    /**
     *
     * @param {string | UdtText} value
     */
    DianExtensionsContent.prototype.setQRCode = function (value) {
        this.attributes.QRCode = value instanceof UnqualifiedDataTypes_1.UdtText ? value : new UnqualifiedDataTypes_1.UdtText(value);
    };
    DianExtensionsContent.prototype.getQRCode = function (raw) {
        if (raw === void 0) { raw = true; }
        return raw ? (this.attributes.QRCode || {}).content : this.attributes.QRCode;
    };
    return DianExtensionsContent;
}(GenericAggregateComponent_1.default));
exports.DianExtensionsContent = DianExtensionsContent;
