// ubl-builder/lib/ubl21/extensionComponents/customExtensionContents/InvoiceControl.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.InvoiceControl = void 0;
var GenericAggregateComponent_1 = require("../../CommonAggregateComponents/GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../../types/UnqualifiedDataTypes");
var CommonAggregateComponents_1 = require("../../CommonAggregateComponents");
var AuthorizedInvoices_1 = require("./AuthorizedInvoices");
var ParamsMap = {
    // ##################################  TODO CAC MISSING ################################################
    invoiceAuthorization: { order: 1, attributeName: 'sts:InvoiceAuthorization', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    authorizationPeriod: { order: 2, attributeName: 'sts:AuthorizationPeriod', min: 0, max: 1, classRef: CommonAggregateComponents_1.PeriodType },
    authorizedInvoices: {
        order: 3,
        attributeName: 'sts:AuthorizedInvoices',
        min: 0,
        max: 1,
        classRef: AuthorizedInvoices_1.AuthorizedInvoices,
    },
};
/**
 * Body of Dian extension content
 */
var InvoiceControl = /** @class */ (function (_super) {
    __extends(InvoiceControl, _super);
    /**     *
     * @param {AllowedParams} content
     * @param {string} name
     */
    function InvoiceControl(content) {
        return _super.call(this, content, ParamsMap, 'sts:InvoiceControl') || this;
    }
    return InvoiceControl;
}(GenericAggregateComponent_1.default));
exports.InvoiceControl = InvoiceControl;
