// ubl-builder/lib/ubl21/CommonAggregateComponents/Attachment.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Attachment = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var ExternalReference_1 = require("./ExternalReference");
var ParamsMap = {
    embeddedDocumentBinaryObject: {
        order: 1,
        attributeName: 'cbc:EmbeddedDocumentBinaryObject',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtBinaryObject,
    },
    externalReference: {
        order: 1,
        attributeName: 'cac:ExternalReference',
        min: 0,
        max: 1,
        classRef: ExternalReference_1.ExternalReference,
    },
};
var Attachment = /** @class */ (function (_super) {
    __extends(Attachment, _super);
    /**
     *
     * @param content children nodes
     */
    function Attachment(content) {
        return _super.call(this, content, ParamsMap, 'cac:Attachment') || this;
    }
    return Attachment;
}(GenericAggregateComponent_1.default));
exports.Attachment = Attachment;
