// ubl-builder/lib/ubl21/CommonAggregateComponents/BillingReference.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BillingReference = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var DocumentReferenceGroup_1 = require("./DocumentReferenceGroup");
var ParamsMap = {
    invoiceDocumentReference: {
        order: 1,
        attributeName: 'cac:InvoiceDocumentReference',
        min: 0,
        max: 1,
        classRef: DocumentReferenceGroup_1.InvoiceDocumentReference,
    },
};
/**
 * A class to define a reference to a billing document
 * More info http://www.datypic.com/sc/ubl21/e-cac_BillingReference.html
 */
var BillingReference = /** @class */ (function (_super) {
    __extends(BillingReference, _super);
    function BillingReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:BillingReference') || this;
    }
    /**
     *
     * @param value
     */
    BillingReference.prototype.setInvoiceDocumentReference = function (value) {
        if (value instanceof DocumentReferenceGroup_1.InvoiceDocumentReference) {
            this.attributes.invoiceDocumentReference = value;
        }
        else {
            throw new Error('this action is not suported yet');
        }
        return this;
    };
    return BillingReference;
}(GenericAggregateComponent_1.default));
exports.BillingReference = BillingReference;
