// ubl-builder/lib/ubl21/CommonAggregateComponents/CorporateRegistrationScheme.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CorporateRegistrationScheme = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var AddressTypeGroup_1 = require("./AddressTypeGroup");
/*
    1  cbc:ID [0..1]    An identifier for this registration scheme.
    2  cbc:Name [0..1]    The name of this registration scheme.
    3  cbc:CorporateRegistrationTypeCode [0..1]    A code signifying the type of this registration scheme.
    4  cac:JurisdictionRegionAddress [0..*]    A geographic area in which this registration scheme applies.
*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    name: { order: 2, attributeName: 'cbc:Name', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtName },
    corporateRegistrationTypeCode: {
        order: 3,
        attributeName: 'cbc:CorporateRegistrationTypeCode',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtCode,
    },
    jurisdictionRegionAddresses: {
        order: 4,
        attributeName: 'cac:JurisdictionRegionAddress',
        min: 0,
        max: undefined,
        classRef: AddressTypeGroup_1.JurisdictionRegionAddress,
    },
};
/**
 *
 */
var CorporateRegistrationScheme = /** @class */ (function (_super) {
    __extends(CorporateRegistrationScheme, _super);
    function CorporateRegistrationScheme(content) {
        return _super.call(this, content, ParamsMap, 'cac:CorporateRegistrationScheme') || this;
    }
    /**
     *
     * @param value
     */
    CorporateRegistrationScheme.prototype.addJurisdictionRegionAddress = function (value) {
        if (!this.attributes.jurisdictionRegionAddresses) {
            this.attributes.jurisdictionRegionAddresses = [];
        }
        if (value instanceof AddressTypeGroup_1.JurisdictionRegionAddress) {
            this.attributes.jurisdictionRegionAddresses.push(value);
        }
        else {
            throw new Error('value must to be instance of JurisdictionRegionAddress');
        }
    };
    return CorporateRegistrationScheme;
}(GenericAggregateComponent_1.default));
exports.CorporateRegistrationScheme = CorporateRegistrationScheme;
