// ubl-builder/lib/ubl21/CommonAggregateComponents/CustomerPartyTypeGroup.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountingCustomerParty = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var PartyTypeGroup_1 = require("./PartyTypeGroup");
var ContactTypeGroup_1 = require("./ContactTypeGroup");
/*

  1  cbc:CustomerAssignedAccountID [0..1]    An identifier for the customer's account, assigned by the customer itself.
  2  cbc:SupplierAssignedAccountID [0..1]    An identifier for the customer's account, assigned by the supplier.
  3  cbc:AdditionalAccountID [0..*]    An identifier for the customer's account, assigned by a third party.
  4  cac:Party [0..1]    The customer party itself.
  5  cac:DeliveryContact [0..1]    A customer contact for deliveries.
  6  cac:AccountingContact [0..1]    A customer contact for accounting.
  7  cac:BuyerContact [0..1]    A customer contact for purchasing.

*/
var ParamsMap = {
    customerAssignedAccountID: {
        order: 1,
        attributeName: 'cbc:CustomerAssignedAccountID',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    SupplierAssignedAccountID: {
        order: 2,
        attributeName: 'cbc:SupplierAssignedAccountID',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    additionalAccountIDs: {
        order: 3,
        attributeName: 'cbc:AdditionalAccountID',
        min: 0,
        max: undefined,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    party: { order: 4, attributeName: 'cac:Party', min: 0, max: 1, classRef: PartyTypeGroup_1.Party },
    deliveryContact: { order: 5, attributeName: 'cac:DeliveryContact', min: 0, max: 1, classRef: ContactTypeGroup_1.DeliveryContact },
    accountingContact: { order: 6, attributeName: 'cac:AccountingContact', min: 0, max: 1, classRef: ContactTypeGroup_1.AccountingContact },
    buyerContact: { order: 7, attributeName: 'cac:BuyerContact', min: 0, max: 1, classRef: ContactTypeGroup_1.BuyerContact },
};
/**
 *
 */
var CustomerParty = /** @class */ (function (_super) {
    __extends(CustomerParty, _super);
    function CustomerParty(content) {
        return _super.call(this, content, ParamsMap, 'cac:CustomerParty') || this;
    }
    /**
     * @returns {Party} The supplier party itself Information
     */
    CustomerParty.prototype.getParty = function () {
        return this.attributes.party;
    };
    return CustomerParty;
}(GenericAggregateComponent_1.default));
exports.AccountingCustomerParty = CustomerParty;
