// ubl-builder/lib/ubl21/CommonAggregateComponents/DeliveryUnitTypeGroup.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaximumDeliveryUnit = exports.MinimumDeliveryUnit = exports.DeliveryUnit = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
/*

cbc:BatchQuantity [1..1]    The quantity of ordered Items that constitutes a batch for delivery purposes.
cbc:ConsumerUnitQuantity [0..1]    The quantity of units in the Delivery Unit expressed in the units used by the consumer.
cbc:HazardousRiskIndicator [0..1]    An indication that the transported goods are subject to an international regulation concerning the carriage of dangerous goods (true) or not (false).

*/
var ParamsMap = {
    batchQuantity: { order: 1, attributeName: 'cbc:BatchQuantity', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtQuantity },
    consumerUnitQuantity: { order: 1, attributeName: 'cbc:ConsumerUnitQuantity', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtQuantity },
    hazardousRiskIndicator: {
        order: 1,
        attributeName: 'cbc:HazardousRiskIndicator',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtIndicator,
    },
};
/**
 *
 */
var DeliveryUnitType = /** @class */ (function (_super) {
    __extends(DeliveryUnitType, _super);
    function DeliveryUnitType(content) {
        return _super.call(this, content, ParamsMap, 'cac:DeliveryUnitType') || this;
    }
    return DeliveryUnitType;
}(GenericAggregateComponent_1.default));
exports.DeliveryUnit = DeliveryUnitType;
exports.MinimumDeliveryUnit = DeliveryUnitType;
exports.MaximumDeliveryUnit = DeliveryUnitType;
