// ubl-builder/lib/ubl21/CommonAggregateComponents/DeliveryTerms.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeliveryTerms = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var UdtAmount_1 = require("../types/UnqualifiedDataTypes/UdtAmount");
var LocationTypeGroup_1 = require("./LocationTypeGroup");
/*

  1    cbc:ID [0..1]    An identifier for this description of delivery terms.
  2    cbc:SpecialTerms [0..*]    A description of any terms or conditions relating to the delivery items.
  3    cbc:LossRiskResponsibilityCode [0..1]    A code that identifies one of various responsibilities for loss risk in the execution of the delivery.
  4    cbc:LossRisk [0..*]    A description of responsibility for risk of loss in execution of the delivery, expressed as text.
  5    cbc:Amount [0..1]    The monetary amount covered by these delivery terms.
  6    cac:DeliveryLocation [0..1]    The location for the contracted delivery.
  7    cac:AllowanceCharge [0..1]    An allowance or charge covered by these delivery terms.

*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    specialTerms: { order: 2, attributeName: 'cbc:SpecialTerms', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    lossRiskResponsibilityCode: {
        order: 3,
        attributeName: 'cbc:LossRiskResponsibilityCode',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtCode,
    },
    lossRisks: { order: 4, attributeName: 'cbc:LossRisk', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    amount: { order: 5, attributeName: 'cbc:Amount', min: 0, max: 1, classRef: UdtAmount_1.UdtAmount },
    deliveryLocation: { order: 6, attributeName: 'cac:DeliveryLocation', min: 0, max: 1, classRef: LocationTypeGroup_1.DeliveryLocation },
};
/**
 *
 */
var DeliveryTerms = /** @class */ (function (_super) {
    __extends(DeliveryTerms, _super);
    /**
     * @param {AllowedParams} content
     * @param {string} name
     */
    function DeliveryTerms(content) {
        return _super.call(this, content, ParamsMap, 'cac:DeliveryTerms') || this;
    }
    return DeliveryTerms;
}(GenericAggregateComponent_1.default));
exports.DeliveryTerms = DeliveryTerms;
