// ubl-builder/lib/ubl21/CommonAggregateComponents/DocumentReferenceGroup.js
"use strict";
/* tslint:disable:max-classes-per-file */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdditionalDocumentReference = exports.ContractDocumentReference = exports.OriginatorDocumentReference = exports.StatementDocumentReference = exports.ReceiptDocumentReference = exports.InvoiceDocumentReference = exports.DocumentReference = exports.DespatchDocumentReference = void 0;
/* TODO GANERIC CLASSES */
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
/* TODO GANERIC CLASSES */
var Attachment_1 = require("./Attachment");
var ValidityPeriod_1 = require("./ValidityPeriod");
// const { ValidityPeriod, ValidityPeriodParams } = require("./ValidityPeriod");
// const { IssuerParty, IssuerPartyParams } = require("./PartyTypeGroup");
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    copyIndicator: { order: 2, attributeName: 'cbc:CopyIndicator', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIndicator },
    uuid: { order: 3, attributeName: 'cbc:UUID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    issueDate: { order: 4, attributeName: 'cbc:IssueDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    issueTime: { order: 5, attributeName: 'cbc:IssueTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    documentTypeCode: { order: 6, attributeName: 'cbc:DocumentTypeCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    documentType: { order: 7, attributeName: 'cbc:DocumentType', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    xPath: { order: 8, attributeName: 'cbc:XPath', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    languageID: { order: 9, attributeName: 'cbc:LanguageID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    localeCode: { order: 10, attributeName: 'cbc:LocaleCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    versionID: { order: 11, attributeName: 'cbc:VersionID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    documentStatusCode: { order: 12, attributeName: 'cbc:DocumentStatusCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    documentDescription: {
        order: 13,
        attributeName: 'cbc:documentDescription',
        min: 0,
        max: undefined,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    attachment: { order: 14, attributeName: 'cac:Attachment', min: 0, max: 1, classRef: Attachment_1.Attachment },
    validityPeriod: { order: 15, attributeName: 'cac:PeriodType', min: 0, max: 1, classRef: ValidityPeriod_1.ValidityPeriod },
};
var DespatchDocumentReference = /** @class */ (function (_super) {
    __extends(DespatchDocumentReference, _super);
    function DespatchDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:DespatchDocumentReference') || this;
    }
    return DespatchDocumentReference;
}(GenericAggregateComponent_1.default));
exports.DespatchDocumentReference = DespatchDocumentReference;
/**
 * A class to define a reference to an Order.
 */
var DocumentReference = /** @class */ (function (_super) {
    __extends(DocumentReference, _super);
    function DocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:DocumentReference') || this;
    }
    /**
     * @param value
     */
    DocumentReference.prototype.setCopyIndicator = function (value) {
        this.attributes.copyIndicator = value instanceof UnqualifiedDataTypes_1.UdtIndicator ? value : new UnqualifiedDataTypes_1.UdtIndicator(value);
        return this;
    };
    return DocumentReference;
}(GenericAggregateComponent_1.default));
exports.DocumentReference = DocumentReference;
var InvoiceDocumentReference = /** @class */ (function (_super) {
    __extends(InvoiceDocumentReference, _super);
    /**
     * @param content
     */
    function InvoiceDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:InvoiceDocumentReference') || this;
    }
    return InvoiceDocumentReference;
}(GenericAggregateComponent_1.default));
exports.InvoiceDocumentReference = InvoiceDocumentReference;
var ReceiptDocumentReference = /** @class */ (function (_super) {
    __extends(ReceiptDocumentReference, _super);
    /**
     * @param {AllowedParams} content
     */
    function ReceiptDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:ReceiptDocumentReference') || this;
    }
    return ReceiptDocumentReference;
}(GenericAggregateComponent_1.default));
exports.ReceiptDocumentReference = ReceiptDocumentReference;
var StatementDocumentReference = /** @class */ (function (_super) {
    __extends(StatementDocumentReference, _super);
    /**
     * @param {AllowedParams} content
     * @param {string} name
     */
    function StatementDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:StatementDocumentReference') || this;
    }
    return StatementDocumentReference;
}(GenericAggregateComponent_1.default));
exports.StatementDocumentReference = StatementDocumentReference;
var OriginatorDocumentReference = /** @class */ (function (_super) {
    __extends(OriginatorDocumentReference, _super);
    /**
     * @param {AllowedParams} content
     */
    function OriginatorDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:OriginatorDocumentReference') || this;
    }
    return OriginatorDocumentReference;
}(GenericAggregateComponent_1.default));
exports.OriginatorDocumentReference = OriginatorDocumentReference;
var ContractDocumentReference = /** @class */ (function (_super) {
    __extends(ContractDocumentReference, _super);
    /**
     * @param {AllowedParams} content
     */
    function ContractDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:ContractDocumentReference') || this;
    }
    return ContractDocumentReference;
}(GenericAggregateComponent_1.default));
exports.ContractDocumentReference = ContractDocumentReference;
var AdditionalDocumentReference = /** @class */ (function (_super) {
    __extends(AdditionalDocumentReference, _super);
    /**
     * @param {AllowedParams} content
     * @param {string} name
     */
    function AdditionalDocumentReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:AdditionalDocumentReference') || this;
    }
    return AdditionalDocumentReference;
}(GenericAggregateComponent_1.default));
exports.AdditionalDocumentReference = AdditionalDocumentReference;
