// ubl-builder/lib/ubl21/CommonAggregateComponents/ExchangeRateTypeGroup.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PricingExchangeRate = exports.PaymentExchangeRate = exports.ExchangeRate = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var UdtRate_1 = require("../types/UnqualifiedDataTypes/UdtRate");
// const GenericAggregateComponent = require("./GenericAggregateComponent");
// /* TODO GANERIC CLASSES */
// const { UdtCode, UdtIdentifier, UdtDate, UdtText, UdtTime, UdtName, UdtQuantity, UdtRate } = require("../types/UnqualifiedDataTypes");
/* TODO GANERIC CLASSES */
/*

  1  cbc:SourceCurrencyCode [1..1]    The reference currency for this exchange rate; the currency from which the exchange is being made.
  2  cbc:SourceCurrencyBaseRate [0..1]    In the case of a source currency with denominations of small value, the unit base.
  3  cbc:TargetCurrencyCode [1..1]    The target currency for this exchange rate; the currency to which the exchange is being made.
  4  cbc:TargetCurrencyBaseRate [0..1]    In the case of a target currency with denominations of small value, the unit base.
  5  cbc:ExchangeMarketID [0..1]    An identifier for the currency exchange market used as the source of this exchange rate.
  6  cbc:CalculationRate [0..1]    The factor applied to the source currency to calculate the target currency.
  7  cbc:MathematicOperatorCode [0..1]    A code signifying whether the calculation rate is a multiplier or a divisor.
  8  cbc:Date [0..1]    The date on which the exchange rate was established.
  9  cac:ForeignExchangeContract [0..1]    A contract for foreign exchange.

*/
var ParamsMap = {
    sourceCurrencyCode: { order: 1, attributeName: 'cbc:SourceCurrencyCode', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    sourceCurrencyBaseRate: { order: 2, attributeName: 'cbc:SourceCurrencyBaseRate', min: 0, max: 1, classRef: UdtRate_1.UdtRate },
    targetCurrencyCode: { order: 3, attributeName: 'cbc:TargetCurrencyCode', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    targetCurrencyBaseRate: { order: 4, attributeName: 'cbc:TargetCurrencyBaseRate', min: 0, max: 1, classRef: UdtRate_1.UdtRate },
    exchangeMarketID: { order: 5, attributeName: 'cbc:ExchangeMarketID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    calculationRate: { order: 6, attributeName: 'cbc:CalculationRate', min: 0, max: 1, classRef: UdtRate_1.UdtRate },
    mathematicOperatorCode: { order: 7, attributeName: 'cbc:MathematicOperatorCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    date: { order: 8, attributeName: 'cbc:Date', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
};
/**
 *
 */
var ExchangeRate = /** @class */ (function (_super) {
    __extends(ExchangeRate, _super);
    function ExchangeRate(content) {
        return _super.call(this, content, ParamsMap, 'cac:ExchangeRate') || this;
    }
    return ExchangeRate;
}(GenericAggregateComponent_1.default));
exports.ExchangeRate = ExchangeRate;
exports.PaymentExchangeRate = ExchangeRate;
exports.PricingExchangeRate = ExchangeRate;
