// ubl-builder/lib/ubl21/CommonAggregateComponents/ExternalReference.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExternalReference = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var ParamsMap = {
    URI: { order: 1, attributeName: 'cbc:URI', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    documentHash: { order: 2, attributeName: 'cbc:DocumentHash', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    hashAlgorithmMethod: { order: 3, attributeName: 'cbc:HashAlgorithmMethod', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    expiryDate: { order: 4, attributeName: 'cbc:ExpiryDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    expiryTime: { order: 5, attributeName: 'cbc:ExpiryTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    mimeCode: { order: 6, attributeName: 'cbc:MimeCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    formatCode: { order: 7, attributeName: 'cbc:FormatCode ', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    encodingCode: { order: 8, attributeName: 'cbc:EncodingCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    characterSetCode: { order: 9, attributeName: 'cbc:CharacterSetCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    fileName: { order: 10, attributeName: 'cbc:FileName', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtName },
    description: { order: 11, attributeName: 'cbc:Description', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
};
// const GenericAggregateComponent = require("./GenericAggregateComponent");
var ExternalReference = /** @class */ (function (_super) {
    __extends(ExternalReference, _super);
    /**
     *
     * @param content
     */
    function ExternalReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:ExternalReference') || this;
    }
    return ExternalReference;
}(GenericAggregateComponent_1.default));
exports.ExternalReference = ExternalReference;
