// ubl-builder/lib/ubl21/CommonAggregateComponents/GenericAggregateComponent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var builder = require("xmlbuilder");
/**
 * Generic class to avoid repeat several code in all CommonAggregateComponent files
 */
var GenericAggregateComponent = /** @class */ (function () {
    /**
     *
     * @param content component content
     * @param paramsMap Params Map
     * @param [name="GenericAggregateComponent"] Class name
     */
    function GenericAggregateComponent(content, paramsMap, name) {
        if (name === void 0) { name = 'GenericAggregateComponent'; }
        // private classRefName: string;
        this.paramsMap = {};
        this.attributes = {};
        // this.classRefName = name;
        this.paramsMap = paramsMap;
        this.assignContent(content);
    }
    GenericAggregateComponent.prototype.parseToJson = function () {
        var _this = this;
        var jsonResponse = {};
        Object.keys(this.paramsMap)
            .filter(function (attkey) { return _this.attributes[attkey]; })
            .forEach(function (attKey) {
            var _a = _this.paramsMap[attKey], attributeName = _a.attributeName, max = _a.max;
            if (Array.isArray(_this.attributes[attKey]) && max !== undefined) {
                throw new Error('array given and max is defined validate structure');
            }
            var attType = typeof _this.attributes[attKey];
            jsonResponse[attributeName] = Array.isArray(_this.attributes[attKey])
                ? _this.attributes[attKey].map(function (e) { return e.parseToJson(); })
                : _this.attributes[attKey].parseToJson();
        });
        return jsonResponse;
    };
    GenericAggregateComponent.prototype.assignContent = function (content) {
        var _this = this;
        // console.log("CONTENT: ", content);
        Object.keys(content || {})
            .filter(function (att) { return content[att] != null; })
            .forEach(function (att) {
            var AbstractClass = _this.paramsMap[att].classRef;
            if (!AbstractClass) {
                throw new Error('classRef is required');
            }
            if (Array.isArray(content[att])) {
                _this.attributes[att] = content[att].map(function (subItem) {
                    return subItem instanceof AbstractClass ? subItem : new AbstractClass(subItem.content, subItem.attributes);
                });
            }
            else {
                var childContent = ['boolean', 'string', 'number'].includes(typeof content[att])
                    ? content[att]
                    : content[att].content;
                var childAttributes = content[att].attributes || {};
                _this.attributes[att] =
                    content[att] instanceof AbstractClass ? content[att] : new AbstractClass(childContent, childAttributes);
            }
        });
    };
    /**
     * @param {boolean} [pretty=true] true for print pretty. true by default
     * @param {boolean} [headless=false] false for print pretty. true by default
     *
     */
    GenericAggregateComponent.prototype.getAsXml = function (pretty, headless) {
        if (pretty === void 0) { pretty = true; }
        if (headless === void 0) { headless = false; }
        var xmlRef = this.parseToJson();
        var xml = builder.create(xmlRef, { encoding: 'UTF-8', standalone: false, headless: headless }).end({ pretty: pretty });
        return xml; // console.log(xml);
    };
    /**
     * @param {boolean} [deep=false] true for deep print
     */
    GenericAggregateComponent.prototype.getAsJson = function (deep) {
        if (deep === void 0) { deep = false; }
        return this.parseToJson();
    };
    return GenericAggregateComponent;
}());
exports.default = GenericAggregateComponent;
