// ubl-builder/lib/ubl21/CommonAggregateComponents/LineReferenceTypeGroup.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestLineReference = exports.QuotationLineReference = exports.ParentDocumentLineReference = exports.DependentLineReference = exports.CatalogueLineReference = exports.CallForTendersLineReference = exports.ReceiptLineReference = exports.DespatchLineReference = exports.LineReference = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var DocumentReferenceGroup_1 = require("./DocumentReferenceGroup");
/* TODO GANERIC CLASSES */
/*
  1  cbc:LineID [1..1] Identifies the referenced line in the document.
  2  cbc:UUID [0..1]   A universally unique identifier for this line reference.
  3  cbc:LineStatusCode [0..1]    A code signifying the status of the referenced line with respect to its original state.
  4  cac:DocumentReference [0..1]    A reference to the document containing the referenced line.
*/
var ParamsMap = {
    lineID: { order: 1, attributeName: 'cbc:LineID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    uuid: { order: 2, attributeName: 'cbc:UUID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    lineStatusCode: { order: 3, attributeName: 'cbc:LineStatusCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    documentReference: { order: 4, attributeName: 'cac:DocumentReference', min: 0, max: 1, classRef: DocumentReferenceGroup_1.DocumentReference },
};
var LineReference = /** @class */ (function (_super) {
    __extends(LineReference, _super);
    function LineReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:LineReference') || this;
    }
    return LineReference;
}(GenericAggregateComponent_1.default));
exports.LineReference = LineReference;
exports.DespatchLineReference = LineReference;
exports.ReceiptLineReference = LineReference;
exports.CallForTendersLineReference = LineReference;
exports.CatalogueLineReference = LineReference;
exports.DependentLineReference = LineReference;
exports.ParentDocumentLineReference = LineReference;
exports.QuotationLineReference = LineReference;
exports.RequestLineReference = LineReference;
