// ubl-builder/lib/ubl21/CommonAggregateComponents/OrderReference.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderReference = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var DocumentReferenceGroup_1 = require("./DocumentReferenceGroup");
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    salesOrderID: { order: 2, attributeName: 'cbc:SalesOrderID', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    copyIndicator: { order: 3, attributeName: 'cbc:CopyIndicator', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIndicator },
    uuid: { order: 4, attributeName: 'cbc:UUID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    issueDate: { order: 5, attributeName: 'cbc:IssueDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    issueTime: { order: 6, attributeName: 'cbc:IssueTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    customerReference: { order: 7, attributeName: 'cbc:CustomerReference', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    orderTypeCode: { order: 8, attributeName: 'cbc:OrderTypeCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    documentReference: { order: 9, attributeName: 'cac:DocumentReference', min: 0, max: 1, classRef: DocumentReferenceGroup_1.DocumentReference },
};
/**
 * http://www.datypic.com/sc/ubl21/e-cac_OrderReference.html
 */
var OrderReference = /** @class */ (function (_super) {
    __extends(OrderReference, _super);
    function OrderReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:OrderReference') || this;
    }
    OrderReference.prototype.setsalesOrderID = function (value) {
        this.attributes.salesOrderID = value instanceof UnqualifiedDataTypes_1.UdtIdentifier ? value : new UnqualifiedDataTypes_1.UdtIdentifier(value);
        return this;
    };
    return OrderReference;
}(GenericAggregateComponent_1.default));
exports.OrderReference = OrderReference;
