// ubl-builder/lib/ubl21/CommonAggregateComponents/OrderLineReference.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderLineReference = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var OrderReference_1 = require("./OrderReference");
// const GenericAggregateComponent = require("./GenericAggregateComponent");
// /* TODO GANERIC CLASSES */
// const { UdtCode, UdtIdentifier, UdtDate, UdtText, UdtTime, UdtName, UdtQuantity } = require("../types/UnqualifiedDataTypes");
// const { OrderReference } = require("./OrderReference");
/* TODO GANERIC CLASSES */
/*

  http://www.datypic.com/sc/ubl21/e-cac_OrderLineReference.html
  1  cbc:LineID [1..1]    An identifier for the referenced order line, assigned by the buyer.
  2  cbc:SalesOrderLineID [0..1]    An identifier for the referenced order line, assigned by the seller.
  3  cbc:UUID [0..1]    A universally unique identifier for this order line reference.
  4  cbc:LineStatusCode [0..1]    A code signifying the status of the referenced order line with respect to its original state.
  5  cac:OrderReference [0..1]    A reference to the Order containing the referenced order line.

*/
var ParamsMap = {
    lineID: { order: 1, attributeName: 'cbc:LineID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    salesOrderLineID: { order: 2, attributeName: 'cbc:SalesOrderLineID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    uuid: { order: 3, attributeName: 'cbc:UUID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    lineStatusCode: { order: 4, attributeName: 'cbc:LineStatusCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    orderReference: { order: 5, attributeName: 'cac:OrderReference', min: 0, max: 1, classRef: OrderReference_1.OrderReference },
};
/**
 *
 */
var OrderLineReference = /** @class */ (function (_super) {
    __extends(OrderLineReference, _super);
    function OrderLineReference(content) {
        return _super.call(this, content, ParamsMap, 'cac:OrderLineReference') || this;
    }
    return OrderLineReference;
}(GenericAggregateComponent_1.default));
exports.OrderLineReference = OrderLineReference;
