// ubl-builder/lib/ubl21/CommonAggregateComponents/PartyTaxScheme.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartyTaxScheme = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var AddressTypeGroup_1 = require("./AddressTypeGroup");
var TaxScheme_1 = require("./TaxScheme");
// const GenericAggregateComponent = require("./GenericAggregateComponent");
// /* TODO GANERIC CLASSES */
// const { UdtCode, UdtIdentifier, UdtDate, UdtText, UdtTime, UdtName } = require("../types/UnqualifiedDataTypes");
// const { RegistrationAddress } = require("./AddressTypeGroup");
// const { TaxScheme, TaxSchemeParams } = require("./TaxScheme");
/* TODO GANERIC CLASSES */
/*
  http://www.datypic.com/sc/ubl21/t-cac_PartyTaxSchemeType.html
  1. cbc:RegistrationName [0..1]    The name of the party as registered with the relevant fiscal authority.
  2. cbc:CompanyID [0..1]    An identifier for the party assigned for tax purposes by the taxation authority.
  3. cbc:TaxLevelCode [0..1]    A code signifying the tax level applicable to the party within this taxation scheme.
  4. cbc:ExemptionReasonCode [0..1]    A reason for the party's exemption from tax, expressed as a code.
  5. cbc:ExemptionReason [0..*]    A reason for the party's exemption from tax, expressed as text.
  6. cac:RegistrationAddress [0..1]    The address of the party as registered for tax purposes.
  7. cac:TaxScheme [1..1]    The taxation scheme applicable to the party.
*/
var ParamsMap = {
    registrationName: { order: 1, attributeName: 'cbc:RegistrationName', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtName },
    companyID: { order: 2, attributeName: 'cbc:CompanyID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    taxLevelCode: { order: 3, attributeName: 'cbc:TaxLevelCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    exemptionReasonCode: { order: 4, attributeName: 'cbc:ExemptionReasonCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    exemptionReason: { order: 5, attributeName: 'cbc:ExemptionReason', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    registrationAddress: {
        order: 6,
        attributeName: 'cac:RegistrationAddress',
        min: 0,
        max: 1,
        classRef: AddressTypeGroup_1.RegistrationAddress,
    },
    taxScheme: { order: 7, attributeName: 'cac:TaxScheme', min: 1, max: 1, classRef: TaxScheme_1.TaxScheme },
};
/**
 * A class to describe a taxation scheme applying to a party.
 * More info http://www.datypic.com/sc/ubl21/t-cac_PartyTaxSchemeType.html
 */
var PartyTaxScheme = /** @class */ (function (_super) {
    __extends(PartyTaxScheme, _super);
    function PartyTaxScheme(content) {
        return _super.call(this, content, ParamsMap, 'cac:PartyTaxScheme') || this;
    }
    /**
     *
     * @param {boolean} [raw=true] raw value
     */
    PartyTaxScheme.prototype.getCompanyID = function (raw) {
        if (raw === void 0) { raw = true; }
        return raw ? this.attributes.companyID.content : this.attributes.companyID;
    };
    return PartyTaxScheme;
}(GenericAggregateComponent_1.default));
exports.PartyTaxScheme = PartyTaxScheme;
