// ubl-builder/lib/ubl21/CommonAggregateComponents/PaymentTypeGroup.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrepaidPayment = exports.PaymentType = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var UdtAmount_1 = require("../types/UnqualifiedDataTypes/UdtAmount");
/*
  1  cbc:ID [0..1]    An identifier for this payment.
  2  cbc:PaidAmount [0..1]    The amount of this payment.
  3  cbc:ReceivedDate [0..1]    The date on which this payment was received.
  4  cbc:PaidDate [0..1]    The date on which this payment was made.
  5  cbc:PaidTime [0..1]    The time at which this payment was made.
  6  cbc:InstructionID [0..1]    An identifier for the payment instruction.
*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    paidAmount: { order: 2, attributeName: 'cbc:PaidAmount', min: 0, max: 1, classRef: UdtAmount_1.UdtAmount },
    receivedDate: { order: 3, attributeName: 'cbc:ReceivedDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    paidDate: { order: 4, attributeName: 'cbc:PaidDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    paidTime: { order: 5, attributeName: 'cbc:PaidTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    instructionID: { order: 6, attributeName: 'cbc:InstructionID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
};
/**
 *
 */
var PaymentType = /** @class */ (function (_super) {
    __extends(PaymentType, _super);
    function PaymentType(content) {
        return _super.call(this, content, ParamsMap, 'cac:PaymentType') || this;
    }
    return PaymentType;
}(GenericAggregateComponent_1.default));
exports.PaymentType = PaymentType;
exports.PrepaidPayment = PaymentType;
