// ubl-builder/lib/ubl21/CommonAggregateComponents/PaymentMeans.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMeans = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
/*
  1  cbc:ID [0..1]    An identifier for this means of payment.
  2  cbc:PaymentMeansCode [1..1]    A code signifying the type of this means of payment.
  3  cbc:PaymentDueDate [0..1]    The date on which payment is due for this means of payment.
  4  cbc:PaymentChannelCode [0..1]    A code signifying the payment channel for this means of payment.
  5  cbc:InstructionID [0..1]    An identifier for the payment instruction.
  6  cbc:InstructionNote [0..*]    Free-form text conveying information that is not contained explicitly in other structures.
  7  cbc:PaymentID [0..*]    An identifier for a payment made using this means of payment.
  8  cac:CardAccount [0..1]    A credit card, debit card, or charge card account that constitutes this means of payment.
  9  cac:PayerFinancialAccount [0..1]    The payer's financial account.
  10  cac:PayeeFinancialAccount [0..1]    The payee's financial account.
  11  cac:CreditAccount [0..1]    A credit account associated with this means of payment.
  12  cac:PaymentMandate [0..1]    The payment mandate associated with this means of payment.
  13  cac:TradeFinancing [0..1]    A trade finance agreement applicable to this means of payment.
*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    paymentMeansCode: { order: 2, attributeName: 'cbc:PaymentMeansCode', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    paymentDueDate: { order: 3, attributeName: 'cbc:PaymentDueDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    paymentChannelCode: { order: 4, attributeName: 'cbc:PaymentChannelCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    instructionID: { order: 5, attributeName: 'cbc:InstructionID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    instructionNotes: { order: 6, attributeName: 'cbc:InstructionNote', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    paymentID: { order: 7, attributeName: 'cbc:PaymentID', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
};
/**
 *
 */
var PaymentMeans = /** @class */ (function (_super) {
    __extends(PaymentMeans, _super);
    function PaymentMeans(content) {
        return _super.call(this, content, ParamsMap, 'cac:PaymentMeans') || this;
    }
    return PaymentMeans;
}(GenericAggregateComponent_1.default));
exports.PaymentMeans = PaymentMeans;
