// ubl-builder/lib/ubl21/CommonAggregateComponents/PeriodTypeGroup.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidityPeriod = exports.RequestedDespatchPeriod = exports.EstimatedDespatchPeriod = exports.EstimatedDeliveryPeriod = exports.PromisedDeliveryPeriod = exports.RequestedDeliveryPeriod = exports.InvoicePeriodBasic = exports.PeriodType = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var ParamsMap = {
    startDate: { order: 1, attributeName: 'cbc:StartDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    startTime: { order: 2, attributeName: 'cbc:StartTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    endDate: { order: 3, attributeName: 'cbc:EndDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    endTime: { order: 4, attributeName: 'cbc:EndTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    durationMeasure: { order: 5, attributeName: 'cbc:DurationMeasure', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtMeasure },
    descriptionCode: { order: 6, attributeName: 'cbc:DescriptionCode', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtCode },
    description: { order: 7, attributeName: 'cbc:Description', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
};
// /* eslint max-classes-per-file: ["error", 2] */
/* tslint:disable:max-classes-per-file */
var PeriodType = /** @class */ (function (_super) {
    __extends(PeriodType, _super);
    function PeriodType(content) {
        return _super.call(this, content, ParamsMap, 'cac:InvoicePeriod') || this;
    }
    return PeriodType;
}(GenericAggregateComponent_1.default));
exports.PeriodType = PeriodType;
exports.RequestedDeliveryPeriod = PeriodType;
exports.PromisedDeliveryPeriod = PeriodType;
exports.EstimatedDeliveryPeriod = PeriodType;
exports.EstimatedDespatchPeriod = PeriodType;
exports.RequestedDespatchPeriod = PeriodType;
exports.ValidityPeriod = PeriodType;
var InvoicePeriodBasic = /** @class */ (function (_super) {
    __extends(InvoicePeriodBasic, _super);
    function InvoicePeriodBasic(startDate, endDate) {
        return _super.call(this, { startDate: startDate, endDate: endDate }) || this;
    }
    return InvoicePeriodBasic;
}(PeriodType));
exports.InvoicePeriodBasic = InvoicePeriodBasic;
