// ubl-builder/lib/ubl21/CommonAggregateComponents/PriceTypeGroup.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Price = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UdtAmount_1 = require("../types/UnqualifiedDataTypes/UdtAmount");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var UdtRate_1 = require("../types/UnqualifiedDataTypes/UdtRate");
var ValidityPeriod_1 = require("./ValidityPeriod");
var AllowanceChargeTypeGroup_1 = require("./AllowanceChargeTypeGroup");
var PriceListTypeGroup_1 = require("./PriceListTypeGroup");
var ExchangeRateTypeGroup_1 = require("./ExchangeRateTypeGroup");
/*

  1    cbc:PriceAmount [1..1]    The amount of the price.
  2    cbc:BaseQuantity [0..1]    The quantity at which this price applies.
  3    cbc:PriceChangeReason [0..*]    A reason for a price change.
  4    cbc:PriceTypeCode [0..1]    The type of price, expressed as a code.
  5    cbc:PriceType [0..1]    The type of price, expressed as text.
  6    cbc:OrderableUnitFactorRate [0..1]    The factor by which the base price unit can be converted to the orderable unit.
  7    cac:ValidityPeriod [0..*]    A period during which this price is valid.
  8    cac:PriceList [0..1]    Information about a price list applicable to this price.
  9    cac:AllowanceCharge [0..*]    An allowance or charge associated with this price.
  10    cac:PricingExchangeRate [0..1]    The exchange rate applicable to this price, if it differs from the exchange rate applicable to the document as a whole.

*/
var ParamsMap = {
    priceAmount: { order: 1, attributeName: 'cbc:PriceAmount', min: 0, max: 1, classRef: UdtAmount_1.UdtAmount },
    baseQuantity: { order: 2, attributeName: 'cbc:BaseQuantity', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtQuantity },
    priceChangeReasons: { order: 3, attributeName: 'cbc:PriceChangeReason', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    priceTypeCode: { order: 4, attributeName: 'cbc:PriceTypeCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    priceType: { order: 5, attributeName: 'cbc:PriceType', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    orderableUnitFactorRate: {
        order: 6,
        attributeName: 'cbc:OrderableUnitFactorRate',
        min: 0,
        max: 1,
        classRef: UdtRate_1.UdtRate,
    },
    validityPeriods: { order: 7, attributeName: 'cac:ValidityPeriod', min: 0, max: 1, classRef: ValidityPeriod_1.ValidityPeriod },
    priceList: { order: 8, attributeName: 'cac:PriceList', min: 0, max: 1, classRef: PriceListTypeGroup_1.PriceList },
    allowanceCharges: {
        order: 9,
        attributeName: 'cac:AllowanceCharge',
        min: 0,
        max: undefined,
        classRef: AllowanceChargeTypeGroup_1.AllowanceCharge,
    },
    pricingExchangeRate: {
        order: 9,
        attributeName: 'cac:PricingExchangeRate',
        min: 0,
        max: 1,
        classRef: ExchangeRateTypeGroup_1.PricingExchangeRate,
    },
};
/**
 *
 */
var Price = /** @class */ (function (_super) {
    __extends(Price, _super);
    function Price(content) {
        return _super.call(this, content, ParamsMap, 'cac:Price') || this;
    }
    Price.prototype.setPriceAmount = function (value) {
        this.attributes.priceAmount = value instanceof UdtAmount_1.UdtAmount ? value : new UdtAmount_1.UdtAmount(value);
    };
    /**
     *
     * @param {boolean} [rawValue=true]
     */
    Price.prototype.getPriceAmount = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.priceAmount.content : this.attributes.priceAmount;
    };
    /**
     *
     * @param { UdtQuantity | string | number} value
     */
    Price.prototype.setBaseQuantity = function (value) {
        this.attributes.baseQuantity = value instanceof UnqualifiedDataTypes_1.UdtQuantity ? value : new UnqualifiedDataTypes_1.UdtQuantity(value);
    };
    /**
     *
     * @param {boolean} rawValue
     */
    Price.prototype.getBaseQuantity = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.baseQuantity.content : this.attributes.baseQuantity;
    };
    return Price;
}(GenericAggregateComponent_1.default));
exports.Price = Price;
