// ubl-builder/lib/ubl21/CommonAggregateComponents/Signature.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Signature = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var ParamsMap = {
    name: { order: 1, attributeName: 'cbc:ID', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    notes: { order: 2, attributeName: 'cbc:Note ', min: 1, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    validationDate: { order: 3, attributeName: 'cbc:ValidationDate', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    validationTime: { order: 4, attributeName: 'cbc:ValidationTime', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    validatorID: { order: 5, attributeName: 'cbc:ValidatorID', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    canonicalizationMethod: { order: 6, attributeName: 'cbc:CanonicalizationMethod', min: 1, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
};
var Signature = /** @class */ (function (_super) {
    __extends(Signature, _super);
    function Signature(content) {
        return _super.call(this, content, ParamsMap, 'cac:Signature') || this;
    }
    return Signature;
}(GenericAggregateComponent_1.default));
exports.Signature = Signature;
