// ubl-builder/lib/ubl21/CommonAggregateComponents/SupplierPartyTypeGroup.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountingSupplierParty = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var PartyTypeGroup_1 = require("./PartyTypeGroup");
/*
    cbc:CustomerAssignedAccountID [0..1]    An identifier for this supplier party, assigned by the customer.
    cbc:AdditionalAccountID [0..*]    An additional identifier for this supplier party.
    cbc:DataSendingCapability [0..1]    Text describing the supplier's ability to send invoice data via a purchase card provider (e.g., VISA, MasterCard, American Express).
    cac:Party [0..1]    The supplier party itself.
    cac:DespatchContact [0..1]    A contact at this supplier party for despatches (pickups).
    cac:AccountingContact [0..1]    A contact at this supplier party for accounting.
    cac:SellerContact [0..1]    The primary contact for this supplier party.
 */
var ParamsMap = {
    customerAssignedAccountID: {
        order: 1,
        attributeName: 'cbc:CustomerAssignedAccountID',
        min: 0,
        max: 1,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    additionalAccountIDs: {
        order: 2,
        attributeName: 'cbc:AdditionalAccountID',
        min: 0,
        max: undefined,
        classRef: UnqualifiedDataTypes_1.UdtIdentifier,
    },
    dataSendingCapability: { order: 3, attributeName: 'cbc:DataSendingCapability', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    party: { order: 4, attributeName: 'cac:Party', min: 0, max: 1, classRef: PartyTypeGroup_1.Party },
};
var AccountingSupplierParty = /** @class */ (function (_super) {
    __extends(AccountingSupplierParty, _super);
    function AccountingSupplierParty(content) {
        return _super.call(this, content, ParamsMap, 'cac:AccountingSupplierParty') || this;
    }
    AccountingSupplierParty.prototype.setParty = function (value) {
        this.attributes.party = value instanceof PartyTypeGroup_1.Party ? value : new PartyTypeGroup_1.Party(value);
    };
    /**
     * @returns {Party} The supplier party itself Information
     */
    AccountingSupplierParty.prototype.getParty = function () {
        return this.attributes.party;
    };
    AccountingSupplierParty.prototype.addAdditionalAccountID = function (value) {
        if (!this.attributes.additionalAccountIDs) {
            this.attributes.additionalAccountIDs = [];
        }
        var itemToPush = value instanceof UnqualifiedDataTypes_1.UdtIdentifier ? value : new UnqualifiedDataTypes_1.UdtIdentifier(value);
        this.attributes.additionalAccountIDs.push(itemToPush);
    };
    return AccountingSupplierParty;
}(GenericAggregateComponent_1.default));
exports.AccountingSupplierParty = AccountingSupplierParty;
