// ubl-builder/lib/ubl21/CommonAggregateComponents/TaxCategoryTypeGroup.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxCategory = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var UdtPercent_1 = require("../types/UnqualifiedDataTypes/UdtPercent");
var UdtAmount_1 = require("../types/UnqualifiedDataTypes/UdtAmount");
var TaxScheme_1 = require("./TaxScheme");
/*

  1  cbc:ID [0..1]    An identifier for this tax category.
  2  cbc:Name [0..1]    The name of this tax category.
  3  cbc:Percent [0..1]    The tax rate for this category, expressed as a percentage.
  4  cbc:BaseUnitMeasure [0..1]    A Unit of Measures used as the basic for the tax calculation applied at a certain rate per unit.
  5  cbc:PerUnitAmount [0..1]    Where a tax is applied at a certain rate per unit, the rate per unit applied.
  6  cbc:TaxExemptionReasonCode [0..1]    The reason for tax being exempted, expressed as a code.
  7  cbc:TaxExemptionReason [0..*]    The reason for tax being exempted, expressed as text.
  8  cbc:TierRange [0..1]    Where a tax is tiered, the range of taxable amounts that determines the rate of tax applicable to this tax category.
  9  cbc:TierRatePercent [0..1]    Where a tax is tiered, the tax rate that applies within the specified range of taxable amounts for this tax category.
  10  cac:TaxScheme [1..1]    The taxation scheme within which this tax category is defined.

*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    name: { order: 2, attributeName: 'cbc:Name', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtName },
    percent: { order: 3, attributeName: 'cbc:Percent', min: 0, max: 1, classRef: UdtPercent_1.UdtPercent },
    baseUnitMeasure: { order: 4, attributeName: 'cbc:BaseUnitMeasure', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtMeasure },
    perUnitAmount: { order: 5, attributeName: 'cbc:PerUnitAmount', min: 0, max: 1, classRef: UdtAmount_1.UdtAmount },
    taxExemptionReasonCode: { order: 6, attributeName: 'cbc:TaxExemptionReasonCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    taxExemptionReason: { order: 7, attributeName: 'cbc:TaxExemptionReason', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
    tierRange: { order: 8, attributeName: 'cbc:TierRange', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    tierRatePercent: { order: 9, attributeName: 'cbc:TierRatePercent', min: 0, max: 1, classRef: UdtPercent_1.UdtPercent },
    taxScheme: { order: 10, attributeName: 'cac:TaxScheme', min: 0, max: 1, classRef: TaxScheme_1.TaxScheme },
};
/**
 *
 */
var TaxCategoryType = /** @class */ (function (_super) {
    __extends(TaxCategoryType, _super);
    function TaxCategoryType(content) {
        return _super.call(this, content, ParamsMap, 'cac:TaxCategoryType') || this;
    }
    TaxCategoryType.prototype.setPercent = function (value) {
        this.attributes.percent = value instanceof UdtPercent_1.UdtPercent ? value : new UdtPercent_1.UdtPercent(value);
    };
    /**
     * @returns { string | UdtPercent }
     */
    TaxCategoryType.prototype.getPercent = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.percent.content : this.attributes.percent;
    };
    TaxCategoryType.prototype.getTaxScheme = function () {
        return this.attributes.taxScheme;
    };
    return TaxCategoryType;
}(GenericAggregateComponent_1.default));
exports.TaxCategory = TaxCategoryType;
