// ubl-builder/lib/ubl21/CommonAggregateComponents/TaxScheme.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxScheme = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var AddressTypeGroup_1 = require("./AddressTypeGroup");
/*
  cbc:ID [0..1]    An identifier for this taxation scheme.
  cbc:Name [0..1]    The name of this taxation scheme.
  cbc:TaxTypeCode [0..1]    A code signifying the type of tax.
  cbc:CurrencyCode [0..1]    A code signifying the currency in which the tax is collected and reported.
  cac:JurisdictionRegionAddress [0..*]    A geographic area in which this taxation scheme applies.
*/
var ParamsMap = {
    id: { order: 1, attributeName: 'cbc:ID', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtIdentifier },
    name: { order: 2, attributeName: 'cbc:Name', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtName },
    taxTypeCode: { order: 3, attributeName: 'cbc:TaxTypeCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    currencyCode: { order: 4, attributeName: 'cbc:CurrencyCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    jurisdictionRegionAddress: {
        order: 5,
        attributeName: 'cac:JurisdictionRegionAddress',
        min: 0,
        max: 1,
        classRef: AddressTypeGroup_1.JurisdictionRegionAddress,
    },
};
/**
 * A class to describe a taxation scheme (e.g., VAT, State tax, County tax).
 */
var TaxScheme = /** @class */ (function (_super) {
    __extends(TaxScheme, _super);
    function TaxScheme(content) {
        return _super.call(this, content, ParamsMap, 'cac:TaxScheme') || this;
    }
    /**
     *
     * @param value
     */
    TaxScheme.prototype.setId = function (value) {
        this.attributes.id = value instanceof UnqualifiedDataTypes_1.UdtIdentifier ? value : new UnqualifiedDataTypes_1.UdtIdentifier(value);
    };
    /**
     *  return Tax's id
     * @param rawValue
     */
    TaxScheme.prototype.getId = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.id.content : this.attributes.id;
    };
    /**
     *
     * @param value
     */
    TaxScheme.prototype.setName = function (value) {
        this.attributes.name = value instanceof UnqualifiedDataTypes_1.UdtName ? value : new UnqualifiedDataTypes_1.UdtName(value);
    };
    /**
     *
     * @param {boolean} rawValue
     */
    TaxScheme.prototype.getName = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.name.content : this.attributes.name;
    };
    return TaxScheme;
}(GenericAggregateComponent_1.default));
exports.TaxScheme = TaxScheme;
