// ubl-builder/lib/ubl21/CommonAggregateComponents/TaxSubtotal.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaxSubtotal = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UdtAmount_1 = require("../types/UnqualifiedDataTypes/UdtAmount");
var UdtNumeric_1 = require("../types/UnqualifiedDataTypes/UdtNumeric");
var UdtPercent_1 = require("../types/UnqualifiedDataTypes/UdtPercent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var TaxCategoryTypeGroup_1 = require("./TaxCategoryTypeGroup");
/*

  1  cbc:TaxableAmount [0..1]    The net amount to which the tax percent (rate) is applied to calculate the tax amount.
  2  cbc:TaxAmount [1..1]    The amount of this tax subtotal.
  3  cbc:CalculationSequenceNumeric [0..1]    The number of this tax subtotal in the sequence of subtotals corresponding to the order in which multiple taxes are applied. If all taxes are applied to the same taxable amount (i.e., their order of application is inconsequential), then CalculationSequenceNumeric is 1 for all tax subtotals applied to a given amount.
  4  cbc:TransactionCurrencyTaxAmount [0..1]    The amount of this tax subtotal, expressed in the currency used for invoicing.
  5  cbc:Percent [0..1]    The tax rate of the tax category applied to this tax subtotal, expressed as a percentage.
  6  cbc:BaseUnitMeasure [0..1]    The unit of measure on which the tax calculation is based
  7  cbc:PerUnitAmount [0..1]    Where a tax is applied at a certain rate per unit, the rate per unit applied.
  8  cbc:TierRange [0..1]    Where a tax is tiered, the range of taxable amounts that determines the rate of tax applicable to this tax subtotal.
  9  cbc:TierRatePercent [0..1]    Where a tax is tiered, the tax rate that applies within a specified range of taxable amounts for this tax subtotal.
  10  cac:TaxCategory [1..1]    The tax category applicable to this subtotal.

*/
var ParamsMap = {
    taxableAmount: { order: 1, attributeName: 'cbc:TaxableAmount', min: 0, max: 1, classRef: UdtAmount_1.UdtAmount },
    taxAmount: { order: 2, attributeName: 'cbc:TaxAmount', min: 1, max: 1, classRef: UdtAmount_1.UdtAmount },
    calculationSequenceNumeric: {
        order: 3,
        attributeName: 'cbc:CalculationSequenceNumeric',
        min: 0,
        max: 1,
        classRef: UdtNumeric_1.UdtNumeric,
    },
    transactionCurrencyTaxAmount: {
        order: 4,
        attributeName: 'cbc:TransactionCurrencyTaxAmount',
        min: 0,
        max: 1,
        classRef: UdtAmount_1.UdtAmount,
    },
    percent: { order: 5, attributeName: 'cbc:Percent', min: 0, max: 1, classRef: UdtPercent_1.UdtPercent },
    baseUnitMeasure: { order: 6, attributeName: 'cbc:BaseUnitMeasure', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtMeasure },
    perUnitAmount: { order: 7, attributeName: 'cbc:PerUnitAmount', min: 0, max: 1, classRef: UdtAmount_1.UdtAmount },
    tierRange: { order: 8, attributeName: 'cbc:TierRange', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtText },
    tierRatePercent: { order: 9, attributeName: 'cbc:TierRatePercent', min: 0, max: 1, classRef: UdtPercent_1.UdtPercent },
    taxCategory: { order: 10, attributeName: 'cac:TaxCategory', min: 1, max: 1, classRef: TaxCategoryTypeGroup_1.TaxCategory },
};
/**
 *
 */
var TaxSubtotal = /** @class */ (function (_super) {
    __extends(TaxSubtotal, _super);
    /**     *
     * @param {AllowedParams} content
     * @param {string} name
     */
    function TaxSubtotal(content) {
        return _super.call(this, content, ParamsMap, 'cac:TaxSubtotal') || this;
    }
    /**
     *
     * @param {boolean} rawValue
     */
    TaxSubtotal.prototype.getTaxableAmount = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.taxableAmount.content : this.attributes.taxableAmount;
    };
    TaxSubtotal.prototype.setTaxableAmount = function (value) {
        this.attributes.taxableAmount = value instanceof UdtAmount_1.UdtAmount ? value : new UdtAmount_1.UdtAmount(value);
    };
    /**
     *
     * @param {boolean} rawValue
     *
     */
    TaxSubtotal.prototype.getTaxAmount = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.taxAmount.content : this.attributes.taxAmount;
    };
    /**
     *
     * @param { UdtAmount | string } value
     */
    TaxSubtotal.prototype.setTaxAmount = function (value) {
        this.attributes.taxAmount = value instanceof UdtAmount_1.UdtAmount ? value : new UdtAmount_1.UdtAmount(value);
    };
    /**
     *
     * @param {*} rawValue
     */
    TaxSubtotal.prototype.getPercent = function (rawValue) {
        if (rawValue === void 0) { rawValue = true; }
        return rawValue ? this.attributes.percent.content : this.attributes.percentage;
    };
    /**
     *
     * @param { UdtPercent | string} value
     */
    TaxSubtotal.prototype.setPercent = function (value) {
        this.attributes.percent = value instanceof UdtPercent_1.UdtPercent ? value : new UdtPercent_1.UdtPercent(value);
    };
    /**
     *
     * @param {TaxCategory} value
     */
    TaxSubtotal.prototype.setTaxCategory = function (value) {
        if (!(value instanceof TaxCategoryTypeGroup_1.TaxCategory))
            throw new Error('Value must be instace of TaxCategory');
        this.attributes.taxCategory = value;
    };
    TaxSubtotal.prototype.getTaxCategory = function () {
        return this.attributes.taxCategory;
    };
    return TaxSubtotal;
}(GenericAggregateComponent_1.default));
exports.TaxSubtotal = TaxSubtotal;
