// ubl-builder/lib/ubl21/CommonAggregateComponents/ValidityPeriod.js
"use strict";
// 'use strict'
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidityPeriod = void 0;
var GenericAggregateComponent_1 = require("./GenericAggregateComponent");
var UnqualifiedDataTypes_1 = require("../types/UnqualifiedDataTypes");
var ParamsMap = {
    startDate: { order: 1, attributeName: 'cbc:StartDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    startTime: { order: 2, attributeName: 'cbc:StartTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    endDate: { order: 3, attributeName: 'cbc:EndDate', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtDate },
    endTime: { order: 4, attributeName: 'cbc:EndTime', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtTime },
    durationMeasure: { order: 5, attributeName: 'cbc:DurationMeasure', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtMeasure },
    descriptionCode: { order: 6, attributeName: 'cbc:DescriptionCode', min: 0, max: 1, classRef: UnqualifiedDataTypes_1.UdtCode },
    description: { order: 7, attributeName: 'cbc:Description', min: 0, max: undefined, classRef: UnqualifiedDataTypes_1.UdtText },
};
var ValidityPeriod = /** @class */ (function (_super) {
    __extends(ValidityPeriod, _super);
    /**
     * @param {AllowedParams} content
     */
    function ValidityPeriod(content) {
        return _super.call(this, content, ParamsMap, 'cac:ValidityPeriod') || this;
    }
    return ValidityPeriod;
}(GenericAggregateComponent_1.default));
exports.ValidityPeriod = ValidityPeriod;
